src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -92,6 +92,32 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 	func showTanScreen() {
 		let tanInputViewModel = TanInputViewModel(
 			coronaTestService: model.coronaTestService,
+			presentOverrideTestNotice: { [weak self] testType, feedbackHandler in
+				
+				let footerViewModel = FooterViewModel(
+					primaryButtonName: AppStrings.ExposureSubmission.OverwriteNotice.primaryButton,
+					isSecondaryButtonHidden: true
+				)
+
+				let overwriteNoticeViewController = TestOverwriteNoticeViewController(
+					testType: testType,
+					didTapPrimaryButton: {
+						self?.navigationController?.dismiss(animated: true) {
+							feedbackHandler(true)
+						}
+					},
+					didTapCloseButton: { [weak self] in
+						self?.navigationController?.dismiss(animated: true) {
+							feedbackHandler(false)
+						}
+					}
+				)
+
+				let footerViewController = FooterViewController(footerViewModel)
+				let topBottomViewController = TopBottomContainerViewController(topController: overwriteNoticeViewController, bottomController: footerViewController)
+				let nc = DismissHandlingNavigationController(rootViewController: topBottomViewController)
+				self?.navigationController?.present(nc, animated: true)
```

Comment:
Is there a reason why it is presented here and not pushed like it is after scanning a second test?

Reply:
It looked very weird to me pushing that view here.

Reply:
UX wanted it to be pushed and the whole submission flow to be dismissed when the close button is pushed.

I think it would be easiest and most consistent with the rest of the coordinator to call `showOverrideTestNotice` here and keep the logic that decides if it needs to be shown also in the coordinator. We would just need to extend the `CoronaTestQRCodeInformation` with the TAN case.

Reply:
Done